wurst/data/buffer/StringBuffer.wurst

```diff
@@ -5,6 +5,29 @@ import LinkedList
 import BufferInterface
 import SafetyChecks
 
+/**
+	This is an abstract class providing an interface to reading/writing chunked
+	string data.
+
+    Typical decode flow:
+        1. Create an instance of an implementing class.
+        2. Push encoded strings using `AbstractStringChunkedBuffer.pushChunk()`
+        3. Call `AbstractStringChunkedBuffer.afterChunkInject()` This will prepare the class for reading.
+        4. Use the `read` methods to extract data.
+
+    Typical encode flow:
+        1. Create an instance of an implementing class.
+        2. Write data using `write` methods.
+        3. Call `AbstractStringChunkedBuffer.beforeChunkExtract()` This will prepare the class for exctracting chunks.
+        4. Use `AbstractStringChunkedBuffer.popChunk()` and `AbstractStringChunkedBuffer.hasChunk()` to extract encoded strings.
+**/
+public abstract class AbstractStringChunkedBuffer extends AbstractBuffer
+	abstract function pushChunk(string chunk)
+	abstract function hasChunk() returns boolean
+	abstract function popChunk() returns string
+	function beforeChunkExtract()
```

Comment:
Do you have any suggestions?

Reply:
> The name of a function is usually a verb or a verb phrase saying what the method does: add, castImmediate. The name should also suggest if the method is mutating the object or returning a new one.
> The names should make it clear what the purpose of the entity is, so it’s best to avoid using meaningless words (Manager, Wrapper etc.) in names.

Thus I would do something like `beginExtraction`